### PR TITLE
Improved rotation vector computation

### PR DESCRIPTION
**Pull request by maxcapodi78**
Improved method get_component_placement_vector that is useful in case of automatic placement of aedb layout.
Minor fixes to be compatible with AEDT 2022R1

### CODE REVIEW

File: pyaedt/edb_core/components.py [hunk 1]
```diff
@@ -447,34 +447,41 @@ def get_component_placement_vector(
...
+        if rotation != 0.0:
+            layinst = mounted_component.GetLayout().GetLayoutInstance()
+            cmpinst = layinst.GetLayoutObjInstance(mounted_component, None)
+            center = cmpinst.GetCenter()
+            center_double = [center.X.ToDouble(), center.Y.ToDouble(), 0]
+            vector_center = GeometryOperators.v_points(center_double, m_pin1_pos_3d)
+            x_v2 = vector_center[0] * math.cos(rotation) + vector_center[1] * math.sin(rotation)
+            y_v2 = -1 * vector_center[0] * math.sin(rotation) + vector_center[1] * math.cos(rotation)
+            new_vector = [x_v2 + center_double[0], y_v2 + center_double[1], 0]
+            vector = [h_pin1_pos[0] - new_vector[0], h_pin1_pos[1] - new_vector[1]]
```
Comment (isaacansys) [inline pyaedt/edb_core/components.py:483]: I am following the logic here, but the tests in test_00_EDB.py should check for the correct result for various cases in addition to just checking that the method returns True.  For example:

- Translation only (rotation should be zero)
- Rotation only (translation should be zero)
- Translation and rotation
- No translation or rotation (ask for component translation onto itself)

File: _unittest_ironpython/run_unittests.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 args_env = os.environ.get("RUN_UNITTESTS_ARGS", "")
 parser = argparse.ArgumentParser()
-parser.add_argument("--test-filter", "-t", default="test_*.py", help="test filter")
+parser.add_argument("--test-filter", "-t", default="test_41*.py", help="test filter")
```
Comment (Alberto-DM) [inline _unittest_ironpython/run_unittests.py:29]:
```suggestion
parser.add_argument("--test-filter", "-t", default="test_*.py", help="test filter")
```
You forgot to reset the filter.

File: pyaedt/edb_core/components.py (hunk 1 above)
Comment (maxcapodi78) [reply pyaedt/edb_core/components.py:486]: I've added an additional test

File: pyaedt/edb_core/components.py [hunk 3]
```diff
@@ -447,34 +447,41 @@ def get_component_placement_vector(
...
-        try:
-            ang1 = math.atan((m_pin2_pos[1] - m_pin1_pos[1]) / ((m_pin2_pos[0] - m_pin1_pos[0])))
-            ang2 = math.atan((h_pin2_pos[1] - h_pin1_pos[1]) / ((h_pin2_pos[0] - h_pin1_pos[0])))
-        except:
-            ang1 = 0.0
-            ang2 = 0.0
-        rotation = ang1 - ang2
         if vector:
             solder_ball_height = self.get_solder_ball_height(mounted_component)
             if solder_ball_height == 0.0:
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/edb_core/components.py]: I've removed default value. it is useless

File: pyaedt/edb_core/components.py [hunk 4]
```diff
@@ -447,34 +447,41 @@ def get_component_placement_vector(
             return False
         if not isinstance(hosting_component, self._edb.Cell.Hierarchy.Component):
             return False
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/edb_core/components.py:449]: I've returned a tuple of 0 to maintain integrity of the unpack

File: pyaedt/edb_core/components.py [hunk 5]
```diff
@@ -447,42 +447,50 @@ def get_component_placement_vector(
...
         if hosting_component_pin1:
             h_pin1 = self._get_edb_pin_from_pin_name(hosting_component, hosting_component_pin1)
             h_pin1_pos = self.get_pin_position(h_pin1)
+            h_pin1_pos_3d = [h_pin1_pos[0], h_pin1_pos[1], 0]
+
         if hosting_component_pin2:
             h_pin2 = self._get_edb_pin_from_pin_name(hosting_component, hosting_component_pin2)
             h_pin2_pos = self.get_pin_position(h_pin2)
+            h_pin2_pos_3d = [h_pin2_pos[0], h_pin2_pos[1], 0]
+        #
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:469]:
```suggestion
```

**Pull request merged**